From foo@baz Tue Apr  9 12:12:43 2002
Date: Wed, 26 Sep 2007 11:12:00 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: driver core: remove subsystem_init()

There is only one user of it, and it is only a wrapper for kset_init().

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/base/class.c    |    2 +-
 include/linux/kobject.h |    1 -
 lib/kobject.c           |    5 -----
 3 files changed, 1 insertion(+), 7 deletions(-)

--- a/drivers/base/class.c
+++ b/drivers/base/class.c
@@ -861,7 +861,7 @@ int __init classes_init(void)
 
 	/* ick, this is ugly, the things we go through to keep from showing up
 	 * in sysfs... */
-	subsystem_init(&class_obj_subsys);
+	kset_init(&class_obj_subsys);
 	if (!class_obj_subsys.kobj.parent)
 		class_obj_subsys.kobj.parent = &class_obj_subsys.kobj;
 	return 0;
--- a/include/linux/kobject.h
+++ b/include/linux/kobject.h
@@ -217,7 +217,6 @@ extern struct kset hypervisor_subsys;
 #define kobj_set_kset_s(obj,subsys) \
 	(obj)->kobj.kset = &(subsys)
 
-extern void subsystem_init(struct kset *);
 extern int __must_check subsystem_register(struct kset *);
 extern void subsystem_unregister(struct kset *);
 
--- a/lib/kobject.c
+++ b/lib/kobject.c
@@ -617,11 +617,6 @@ struct kobject * kset_find_obj(struct ks
 	return ret;
 }
 
-void subsystem_init(struct kset *s)
-{
-	kset_init(s);
-}
-
 int subsystem_register(struct kset *s)
 {
 	return kset_register(s);
